executor/simple.go

```diff
@@ -661,6 +661,12 @@ func (e *SimpleExec) executeRevokeRole(ctx context.Context, s *ast.RevokeRoleStm
 		return errors.Trace(err)
 	}
 	sql := new(strings.Builder)
+	// when an active role of current user is revoked,
+	// it should be removed from activeRoles
```

Comment:
What would happen for other active users with the role being revoked?

Reply:
What has priority to be fixed, is that the `revoke role r1 from u1` revokes the extra rights the `r1` role gives to `u1`, even in currently running sessions (for any statement started after the revoke succeeded). If there are errors or other issues, that has lower impact. Could you please test what would happen for a concurrent u1 session (preferably on another tidb instance)? I'm all good in not fixing those in this bug, but we should at least check and report new bugs, since it is security related.

Reply:
I wonder if the error is not because the u1 user did not set the role? (I at least see it as a bug, but can probably be seen as a different bug that this).
See below that it warns about role `r1`, which I never gave to u1@localhost!
```sql
$ mysql -S /tmp/tidb-4001.sock -u u1 --prompt="tidb> " --comments  --database test -v
...
Your MySQL connection id is 17
Server version: 5.7.25-TiDB-v5.4.0-alpha-121-g724020289 TiDB Server (Apache License 2.0) Community Edition, MySQL 5.7 compatible
...
tidb> show grants;
ERROR 3530 (HY000): `r1`@`%` is not granted to u1@localhost

tidb> set role r1;
ERROR 3530 (HY000): `r1`@`%` is not granted to u1@localhost

tidb> set role r2;
Query OK, 0 rows affected (0,00 sec)

tidb> show grants;
+---------------------------------------------+
| Grants for User                             |
+---------------------------------------------+
| GRANT USAGE ON *.* TO 'u1'@'localhost'      |
| GRANT SELECT ON test.t2 TO 'u1'@'localhost' |
| GRANT 'r2'@'%' TO 'u1'@'localhost'          |
+---------------------------------------------+
3 rows in set (0,00 sec)
```

Users, tables and roles created by:
```sql
tidb> create user u1;
Query OK, 0 rows affected (0,00 sec)

tidb> create user u1@localhost;
Query OK, 0 rows affected (0,01 sec)

tidb> create role r1, r2;
Query OK, 0 rows affected (0,01 sec)

tidb> show tables;
Empty set (0,00 sec)

tidb> create table t1 (a int);
Query OK, 0 rows affected (0,00 sec)

tidb> create table t2 (a int);
Query OK, 0 rows affected (0,00 sec)

tidb> grant select on test.t2 to r2;
Query OK, 0 rows affected (0,01 sec)

tidb> grant select on test.* to r1;
Query OK, 0 rows affected (0,01 sec)

tidb> grant r1 to u1;
Query OK, 0 rows affected (0,01 sec)

tidb> grant r2 to u1@localhost;
Query OK, 0 rows affected (0,01 sec)
```

Reply:
I just tried and revoking a role does still leave the privileges intact for the user, even for new sessions, while revoking the privileges from the role, does remove the privileges granted through the role.

Reply:
> it warns about role r1, which I never gave to u1@localhost!

Does it means that  `revoke r1 from u1` will cause `show grants`  of `u1@localhost` fail? I tried but can't reproduce it, could you please give some advice?

> I just tried and revoking a role for another user does still leave the privileges intact for that user, even for new sessions,

After execute `revoke r1 from u1`, `u1` still can execute `select * from t1;`, but I think it may be another issue to be reported